libs/csm/rest/csm_rest_capacity.py

```diff
@@ -47,32 +46,104 @@ def __init__(self):
         self.cryptogen = SystemRandom()
 
     @RestTestLib.authenticate_and_login
-    def get_capacity_usage(self):
+    def get_capacity_usage(self, endpoint_param=None ,auth_header=None):
         """Get the system capacity usage
 
         :return [obj]: Json response
         """
-        try:
-            # Building request url
-            self.log.info("Reading System Capacity...")
-            endpoint = self.config["capacity_endpoint"]
-            self.log.info("Endpoint for reading capacity is %s", endpoint)
+        if auth_header is None:
+            headers = self.headers
+        else:
+            headers = auth_header
+
+        self.log.info("Reading System Capacity...")
+        endpoint = self.config["capacity_endpoint"]
+        if endpoint_param is not None:
+            endpoint = self.config["capacity_endpoint"] + "/" + endpoint_param
+        self.log.info("Endpoint for reading capacity is %s", endpoint)
+        # Fetching api response
+        response = self.restapi.rest_call(request_type="get", endpoint=endpoint,
+                                          headers=headers)
+        return response
+
+    def get_degraded_capacity(self, auth_header=None):
```

Comment:
Can we rename this to get_capacity_bytecount or get_bytecount, as it is not specific to degraded capacity?

Reply:
it is specific to degraded capacity, As it gives degraded counts. Also this function is used everywhere.

Reply:
My only concern is the confusion it will cause for tests which will first call get_degraded_capacity and then fetch the "healthy" bytecount from the returned response.

If this is not possible, renaming the function that returns the munch will help as well.

Reply:
@apoorva-rao : This is motr view of the system. Not the available capacity as seen by users. It is in degraded context only. There is separate feature for cluster level capacity.

Reply:
I will retain this name for now as it is used. Changed the name for the other function.

Reply:
I will retain this name for now as it is used. Changed the name for the other function.